wordpress-mobile/WordPress-Android
# AppBar refresh - Site Picker, Plans, Activity Log, Themes, Publicize

#12547 needs to be merged first.

This PR updates AppBar design and adds lift on scroll functionality to the following screens (and related nested screens):
- Site Picker
- Plans
- Activity Log (+ added missing scrolling behavior to activity details screen).
- Theme Browser (there is a known issue where AppBar elevation does not update automatically after search (you need to touch content) - I wasn't able to find a fix yet, but it's not too critical, so I look for a solution if I'll have time at the end of the project).
- Publicize

What is lift on scroll? When you scroll content below the AppBar its elevation changes, and as a result shadow appears in light mode, and in Dark Mode color of AppBar gets lighter.

To test:

For all screens:
- Make sure nothing looks out of place :)
- Make sure the toolbar state is restored on configuration change.

Test 1:
- Navigate to **Site Picker** screen
- Scroll the sites list and make sure the lift on scroll works.
- Check search and make sure lift on scroll works as expected.

Test 2:
- Navigate to **Activity Log** screen
- Scroll the content and make sure the lift on scroll works.
- Check the activity details and make sure the lift on scroll works (you might need to rotate your device).

Test 3:
- Navigate to **Plan** screen
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).
- Check the plan details and make sure the lift on scroll works.

Test 4:
- Navigate to **Themes** screen
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).
- Check search and make sure lift on scroll works mostly expected (minus the issue I mentioned at the start).

Test 5:
- Navigate to **Publicize** screen
- Scroll the content and make sure the lift on scroll works (you might need to rotate your device).
- Check publicize buttons settings and make sure the lift on scroll works.
- Check publicize connection WebView screen and make sure the lift on scroll works (where applicable)

PR submission checklist:
